src/rpdk/core/contract/resource_client.py

```diff
@@ -267,9 +267,40 @@ def generate_invalid_create_example(self):
         example = self.invalid_strategy.example()
         return override_properties(example, self._overrides.get("CREATE", {}))
 
+    def get_unique_keys_for_model(self, create_model):
+        return {
+            k: v
+            for k, v in create_model.items()
+            if self.is_property_in_path(k, self.primary_identifier_paths)
+            or self.is_property_in_path(k, self._additional_identifiers_paths)
+        }
+
+    # Disable the no-self-use violations in this function
+    # pylint: disable=no-self-use
+    def is_property_in_path(self, prop, paths):
+        for path in paths:
+            for key in path:
+                if prop == key:
+                    return True
+        return False
+
+    def validate_update_example_keys(self, unique_identifiers, update_example):
+        for primary_identifier in self.primary_identifier_paths:
+            if primary_identifier in self.create_only_paths:
+                primary_key = fragment_list(primary_identifier, "properties")[0]
+                assert update_example[primary_key] == unique_identifiers[primary_key], (
+                    "primaryIdentifier at the time of update must match"
+                    " primaryIdentifier used at the time of create"
```

Comment:
this is what we have in handler contract [`Any createOnlyProperties specified in update handler input MUST NOT be different from their previous state`](https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract.html#resource-type-test-contract-update)